myriad-scheduler/src/main/java/org/apache/myriad/scheduler/ExecutorCommandLineGenerator.java

```diff
@@ -19,11 +19,148 @@
+    if (myriadConfiguration.getFrameworkSuperUser().isPresent()) {
+      cmdLine.append(" sudo ");
+    }
+  }
+
+  protected void appendUserSudo(StringBuilder cmdLine) {
+    if (myriadConfiguration.getFrameworkSuperUser().isPresent()) {
+      cmdLine.append(" sudo -E -u ");
+      cmdLine.append(myriadConfiguration.getFrameworkUser().get());
+      cmdLine.append(" -H ");
+    }
+  }
+
+  public String getConfigurationUrl() {
+    String httpPolicy = yarnConfiguration.get(TaskFactory.YARN_HTTP_POLICY);
+    String address = StringUtils.EMPTY;
+    if (httpPolicy != null && httpPolicy.equals(TaskFactory.YARN_HTTP_POLICY_HTTPS_ONLY)) {
+      address = yarnConfiguration.get(TaskFactory.YARN_RESOURCEMANAGER_WEBAPP_HTTPS_ADDRESS);
+      if (StringUtils.isEmpty(address)) {
+        address = yarnConfiguration.get(TaskFactory.YARN_RESOURCEMANAGER_HOSTNAME) + ":8090";
+      }
+    } else {
+      address = yarnConfiguration.get(TaskFactory.YARN_RESOURCEMANAGER_WEBAPP_ADDRESS);
+      if (StringUtils.isEmpty(address)) {
+        address = yarnConfiguration.get(TaskFactory.YARN_RESOURCEMANAGER_HOSTNAME) + ":8088";
+      }
+    }
+
+    return "http://" + address + "/yarnConfiguration";
+  }
+
+  private static String getFileName(String uri) {
+    int lastSlash = uri.lastIndexOf('/');
+    if (lastSlash == -1) {
+      return uri;
+    } else {
+      String fileName = uri.substring(lastSlash + 1);
+      Preconditions.checkArgument(!Strings.isNullOrEmpty(fileName), "URI should not have a slash at the end");
+      return fileName;
+    }
+  }
+
+  protected String getUser() {
+    if (myriadConfiguration.getFrameworkSuperUser().isPresent()) {
+      return myriadConfiguration.getFrameworkSuperUser().get();
+    } else {
+      return myriadConfiguration.getFrameworkUser().get();
+    }
+  }
 
-  String getConfigurationUrl();
+  protected List<Protos.CommandInfo.URI> getUris() {
+    List<Protos.CommandInfo.URI> uris = new ArrayList<>();
+    if (myriadExecutorConfiguration.getJvmUri().isPresent()) {
+      final String jvmRemoteUri = myriadExecutorConfiguration.getJvmUri().get();
+      LOGGER.info("Getting JRE distribution from:" + jvmRemoteUri);
+      uris.add(Protos.CommandInfo.URI.newBuilder().setValue(jvmRemoteUri).build());
+    }
+    if (myriadExecutorConfiguration.getConfigUri().isPresent()) {
+      String configURI = myriadExecutorConfiguration.getConfigUri().get();
+      LOGGER.info("Getting Hadoop distribution from: {}", configURI);
+      uris.add(Protos.CommandInfo.URI.newBuilder().setValue(configURI).build());
+    }
+    if (myriadExecutorConfiguration.getNodeManagerUri().isPresent()) {
+      //Both FrameworkUser and FrameworkSuperuser to get all of the directory permissions correct.
+      if (!(myriadConfiguration.getFrameworkUser().isPresent() && myriadConfiguration.getFrameworkSuperUser().isPresent())) {
+        LOGGER.warn("Trying to use remote distribution, but frameworkUser and/or frameworkSuperUser not set!" +
```

Comment:
I recommend a more explicit logging message stating that the frameworkUser is not set and what the ramifications are.